Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -2622,6 +2625,48 @@ static WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     fireActionClickedEventIfNeeded(tab);
 }
 
+#if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
+WebExtensionSidebar& WebExtensionContext::defaultSidebar()
+{
+    using IsDefaultSidebar = WebExtensionSidebar::IsDefaultSidebar;
+
+    if (!m_defaultSidebar)
+        m_defaultSidebar = WebExtensionSidebar::create(*this, IsDefaultSidebar::Yes);
+
+    return *m_defaultSidebar;
+}
+
+std::optional<Ref<WebExtensionSidebar>> WebExtensionContext::getSidebar(WebExtensionWindow const& window)
+{
+    if (auto *windowAction = m_sidebarWindowMap.get(window))
+        return *windowAction;
+
+    return std::nullopt;
+}
+
+std::optional<Ref<WebExtensionSidebar>> WebExtensionContext::getSidebar(WebExtensionTab const& tab)
+{
+    if (auto *tabAction = m_sidebarTabMap.get(tab))
+        return *tabAction;
+
+    return std::nullopt;
+}
+
+Ref<WebExtensionSidebar> WebExtensionContext::getOrCreateSidebar(WebExtensionWindow& window)
```

Comment:
If the extension doesn't use sidebars, wouldn't these methods never be called? Just since we deny access to the `sidebarAction` / `sidePanel` properties in `WebExtensionAPINamespace::isPropertyAllowed` based on what's present in the manifest.